skunkworks_crow/src/main/java/org/odk/share/tasks/DownloadJob.java

```diff
@@ -94,26 +103,60 @@ protected Result onRunJob(@NonNull Params params) {
         ((Share) getContext().getApplicationContext()).getAppComponent().inject(this);
 
         initJob(params);
-        rxEventBus.post(receiveForms());
 
         return null;
     }
 
     private void initJob(Params params) {
-        ip = params.getExtras().getString(IP, "");
-        port = params.getExtras().getInt(PORT, -1);
         sbResult = new StringBuilder();
+        boolean isBluetooth = params.getExtras().getBoolean("isBluetooth", true);
+        if (!isBluetooth) {
+            ip = params.getExtras().getString(IP, "");
+            port = params.getExtras().getInt(PORT, -1);
+        } else {
+            targetMacAddress = params.getExtras().getString("mac", null);
+        }
+
+        setupDataStreamsAndReceive(isBluetooth, targetMacAddress);
     }
 
-    private DownloadEvent receiveForms() {
-        Timber.d("Socket " + ip + " " + port);
+    private void setupDataStreamsAndReceive(boolean isBluetooth, String macAddress) {
+        try {
+            Timber.d("Waiting for receiver");
```

Comment:
The log should be like `Waiting for sender` because this is the `DownloadJob`

Reply:
Fixed.